File: src/models/tokenizer/Tokenizer.java
package models.tokenizer;

import java.io.File;
import java.util.LinkedList;
import java.util.Map;
import java.util.Set;

public interface Tokenizer {
    String SPLITTER = "\\s";

    default String cleanText(String text) {
        return text.toLowerCase();
    }

    void tokenizeData(File file, String text, Map<String, Map<File, Integer>> data);

    Set<String> tokenizeQuery(String text);

    LinkedList<String> develop(String token);

}

File: C#/SimpleSearch/models/tokenizer/Tokenizer.cs
using SimpleSearch.models.search;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimpleSearch.models.tokenizer
{
    abstract class Tokenizer
    {
        protected static readonly string SPLITTER = "[|\\s|.|'|:|_|-|?|/|@|<|>|!|(|)|]";

        public string CleanText(string text) => text.ToLower();

        public abstract void tokenizeData(string filePath, string text, IDictionary<string, InvertedIndex> data);

        public abstract ISet<string> TokenizeQuery(string text);

        public abstract List<string> Develope(string token);
    }
}
